src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+  AfterContentChecked,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  Output,
+  OnInit,
+  PLATFORM_ID,
+} from '@angular/core';
+
+import { isPlatformBrowser } from '@angular/common';
+
+import { Observable } from 'rxjs';
+
+@Directive({
+  selector: '[clrKeyFocus]',
+})
+export class ClrKeyFocus implements OnInit, AfterContentChecked {
+  constructor(private elementRef: ElementRef, @Inject(PLATFORM_ID) private platformId: Object) {}
+
+  @Input('clrKeyFocus') direction: 'horizontal' | 'vertical' | 'both';
```

Comment:
Comment #1
I saw the comment below in ngOnInit. 

I _think_ you could also set the default with a setter on the input. Its similar to how other components handle the undefined input do as well:

```typescript
private _direction;
@ Input('clrDirection') // not sure if we want to prefix this or just use vanilla direction
set direction(direction) {
  this._direction = !!direction;
}
```

Comment #2
It seems odd to me to have two different names 'clrKeyFocus' and direction.
Just tossing spagetti on the wall here but: 
- clrKeyFocusDirection : direction 
- clrKeyFocus : keyFocus (in anticipation of some sort of an enum ClrFocusDirection.BOTH)

Reply:
1) will definitely be better, thanks!
 2) just wanted to use the directive attribute for something, but it's really a bit weird; will make a separate input for this

Reply:
1. moved to setter
2. input separated from directive attribute; added enum with direction types

done done